content/java/concepts/treemap/treemap.md

```diff
@@ -0,0 +1,196 @@
+---
+Title: 'TreeMap'
+Description: 'Used to store items as key-value pairs, ensuring items are sorted in a natural key order. The keys and values can be either the same or of different types.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Data Structures'
+  - 'Data Types'
+  - 'Collections'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `TreeMap`, like a [`HashMap`](https://www.codecademy.com/resources/docs/java/HashMap), uses the [`Map`](https://www.codecademy.com/resources/docs/java/Map) interface, which means it stores items as key-value pairs. Unlike a `HashMap`, the `TreeMap` stores its items sorted by the natural ordering of its keys. This is due to the differing means of storage each type of collection uses internally. The `HashMap` stores its keys as hashes for lookup, whereas the `TreeMap` stores its keys in a binary tree structure for lookup. The tradeoff for this natural ordering of the `TreeMap` is that it operates slower than a `HashMap`.
+
+## Syntax
+
+The `TreeMap` class comes from the `java.util` package. Therefore, it must be imported in order to be used:
+
+```java
+import java.util.TreeMap
+
+TreeMap<KeyDatatype, ValueDatatype> myTreeMap = new TreeMap<KeyDatatype, ValueDatatype>();
+```
+
+The `TreeMap` is initialized with two generic types inside angle brackets `< ... >`. The generic data types for `KeyDatatype` and `ValueDatatype` can either be different or the same.
+
+## Keys and Values
+
+A key can be used to uniquely identify a value in a `HashMap`. This allows for efficient data storage and easy access. In the example below, the course names are the keys and the teachers assigned are the values that can be accessed using the corresponding key.
```

Comment:
```suggestion
Keys are used for uniquely identifying a value in a `HashMap`. This allows for efficient data storage and easy access. In the example below, the course names are the keys and the teachers assigned are the values that can be accessed by passing the corresponding key into the `.get()` method.
```

Reply:
Similar thing here. I think that since the `.get()` method is first introduced here, maybe we could refer to it by name here and omit the entire "Accessing an Item" section since that's what this snippet seems to be doing?